### PR TITLE
feat: Rework.

### PR DESCRIPTION
**Pull request by Soap-141**
GitHub Issue: #

## Proposed Changes

- Feature

## Checklist

Please check if your PR fulfills the following requirements:

- [x] Documentation has been added/updated
- [ ] Automated Unit / Integration tests for the changes have been added/updated
- [x] Contains **NO** breaking changes
- [ ] Updated the Changelog
- [ ] Associated with an issue

### CODE REVIEW

File: src/BiometryService.Abstractions/BiometryService.Abstractions.csproj [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
```
Comment (jeanplevesque) [inline src/BiometryService.Abstractions/BiometryService.Abstractions.csproj:1]: You're missing metadata. Check what the other package has.

**Review by jeanplevesque — CHANGES_REQUESTED** [review]

**Review by jeanplevesque — APPROVED** [review]

File: src/BiometryService.Abstractions/BiometryType.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+/// </summary>
+/// <remarks>
+/// It is technically possible that a device has multiple biometric available.
+/// That's the reason why this enum has <see cref="FlagsAttribute"/>.
+/// </remarks>
+[Flags]
+public enum BiometryType : byte
+{
+	/// <summary>
+	///	No biometric identifier.
```
Comment (takla21) [inline src/BiometryService.Abstractions/BiometryType.cs:16]: Why we have that much space?

File: src/BiometryService/BiometryService.UWP.cs [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Windows.Foundation.Collections;
+using Windows.Security.Credentials;
+using Windows.Storage;
+
+namespace BiometryService;
+
+/// <summary>
+/// Windows implementation of <see cref="IBiometryService"/>.
```
Comment (takla21) [inline src/BiometryService/BiometryService.UWP.cs:17]: We should probably specify that it's a UWP implementation

File: src/BiometryService/BiometryService.WinUI.cs [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+using Windows.Foundation.Collections;
+using Windows.Security.Credentials;
+using Windows.Storage;
+
+namespace BiometryService;
+
+/// <summary>
+/// Windows implementation of <see cref="IBiometryService"/>.
```
Comment (takla21) [inline src/BiometryService/BiometryService.WinUI.cs:17]: We should specify it's a WinUI implementation

File: src/BiometryService/BiometryService.Android.cs [hunk 5]
```diff
@@ -0,0 +1,426 @@
...
+	{
+		if (_logger.IsEnabled(LogLevel.Debug))
+		{
+			_logger.LogDebug($"Start authenticating the user biometry.");
+		}
+
+		var authenticateCode = 0;
+		if (Android.OS.Build.VERSION.SdkInt <= Android.OS.BuildVersionCodes.Q)
+		{
+			authenticateCode = _biometricManager.CanAuthenticate(); // TODO Eliminate warning somehow?
```
Comment (takla21) [inline src/BiometryService/BiometryService.Android.cs:184]: Should this todo still be relevent?

Comment (Soap-141) [reply src/BiometryService/BiometryService.Android.cs:184]: Yes, I don't know why we need to check the version. This should be done in the suggestion I made on the .NET backlog.

Comment (takla21) [reply src/BiometryService/BiometryService.Android.cs:184]: Right, if I recall I've added this comment as a reminder to find a way to clear the warning, but I guess we should either remove this comment or reword it to be clearer

**Review by takla21 — APPROVED** [review]
Good job!
